model/src/main/proto/exploration.proto

```diff
@@ -332,9 +350,30 @@ message HelpIndex {
     // have been revealed and there's no other help to provide. The boolean value here has no
     // importance and is always 'true'.
     bool everything_revealed = 3;
+
+    // Indicates this help index corresponds to the index of the next available hint within the hint
+    // list of a state.
+    int32 available_next_hint_index = 4;
+
+    // Indicates this help index corresponds to the index of the last revealed hint within the hint
+    // list of a state.
```

Comment:
I think we're missing some important context in this & the above doc comment. The case above specifically indicates that a new hint is available to be revealed whereas this case represents that all available hints have been revealed. Without that distinction these cases seem like they could be combined into one.